### PR TITLE
Added missing 3d-image enforcements

### PR DESCRIPTION
**Pull request by banilo**
Some functions with mask arguments did not complain about iterables of niimg-like objects.

attempts to address #302

### CODE REVIEW

File: nilearn/input_data/tests/test_base_masker.py [hunk 1]
```diff
@@ -3,12 +3,12 @@
 """
 
 import numpy as np
-from numpy.testing import assert_array_almost_equal
+from numpy.testing import assert_array_almost_equal, assert_raises
```
Comment (lesteve) [inline nilearn/input_data/tests/test_base_masker.py]: Unless I am missing something, assert_raises is not used in test_base_masker.py.

**Comment by banilo** [comment]
Any news on this PR?

**Comment by lesteve** [comment]
> I am slightly worried to the same feature tested at different places of the codebase

@bthirion I am guessing you are you referring to the [concat_img PR](https://github.com/nilearn/nilearn/pull/277)  rather than this one

**Comment by bthirion** [comment]
Well as far as I can see, the tests in test_base_masker.py and test_multi_nifti_masker.py are redundant, aren't they ?

**Comment by lesteve** [comment]
> Well as far as I can see, the tests in test_base_masker.py and test_multi_nifti_masker.py are redundant, aren't they ?

OK I am afraid I am still not getting your point unfortunately. I am going to merge this one since it's been open for a while and it fulfills its original promise. Please do feel free to open a separate issue about the redundancy in the tests.

**Pull request merged**